src/coreclr/gc/gc.cpp

```diff
@@ -44894,10 +44894,21 @@ HRESULT GCHeap::Initialize()
     {
         if (gc_heap::heap_hard_limit)
         {
-            gc_heap::regions_range = 2 * gc_heap::heap_hard_limit;
+            if (gc_heap::heap_hard_limit_oh[soh])
+            {
+                gc_heap::regions_range = gc_heap::heap_hard_limit;
+            }
+            else
+            {
+                // We will initially reserve 5x the configured hard limit and 2x if
+                // large pages are enabled as this is closer to segments
+                gc_heap::regions_range = ((gc_heap::use_large_pages_p) ? (2 * gc_heap::heap_hard_limit)  // large pages
```

Comment:
```suggestion
                gc_heap::regions_range = ((gc_heap::use_large_pages_p) ? (2 * gc_heap::heap_hard_limit)
```

no need to say `// large pages` because that's exactly what `use_large_pages_p` says